src/services/userService.ts

```diff
@@ -0,0 +1,55 @@
+import crypto from 'crypto';
+import { BaseUser, User } from '../types/user.js';
+import { usersArray } from '../storage/localUsersStorage.js';
+
+
+class UserService {
+    private static generateUUID(): string {
+        return crypto.randomUUID();
+    }
+    create(user: BaseUser): User {
+        const uuid = UserService.generateUUID();
+        const newUser = {
+            id: uuid,
+            isDeleted: false,
+            ...user
+        };
+        usersArray.push(newUser);
+        return newUser;
+    }
+    get(id: string): User | undefined {
+        return usersArray.find(user => user.id.includes(id) && !user.isDeleted);
+    }
+    update(userUpdates: BaseUser, id: string): User {
+        const updateInstanceIndex = usersArray.findIndex(user => user.id === id);
+        if (updateInstanceIndex !== -1) {
+            usersArray[updateInstanceIndex] = {
+                id,
+                isDeleted: false,
+                ...userUpdates
+            };
+        }
+        return usersArray[updateInstanceIndex];
+    }
+    delete(id: string): void {
```

Comment:
method must return certain result, especially for the delete logic. In the end you will need to know if user was successfully deleted OR not.
In this case I would recommend to return the deleted record(but this is something that can be agreed by the team.. what and how we should return).